truffle.js

```diff
@@ -1,23 +1,64 @@
 require('babel-register')
 require('babel-polyfill')
 
+const HDWalletProvider = require('truffle-hdwallet-provider')
+const fs = require('fs')
+
+// first read in the secrets.json to get our mnemonic
+let secrets
+let mnemonic
+let infuraProjectID
+if (fs.existsSync('secrets.json')) {
```

Comment:
is this reading from you root? confused how this works

Reply:
yes, it's reading from a gitignored `secrets.json` file at the root of the project. if you want to deploy on rinkeby you add this file with the following format:
```
{
  "mnemonic": "<YOUR_MNEMONIC_SEED>",
  "infuraProjectID": "<YOUR_INFURA_PROJECT_ID>"
}
```

infura requires a project ID now FYI so you'll have to create one through their ultra-centralized web service so they can monitor what we're doing on rinkeby and mainnet and tie it back to our human identities